crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -423,14 +419,27 @@ impl CumulocityConverter {
         entity_topic_id: &EntityTopicId,
     ) -> Result<Topic, ConversionError> {
         let entity = self.entity_store.try_get(entity_topic_id)?;
-
-        let mut ancestors_external_ids =
-            self.entity_store.ancestors_external_ids(entity_topic_id)?;
-        ancestors_external_ids.insert(0, entity.external_id.as_ref().into());
-        Ok(publish_topic_from_ancestors(
-            &ancestors_external_ids,
+        let topic = C8yTopic::smartrest_response_topic(
+            &entity.external_id,
+            &entity.r#type,
             &self.config.bridge_config.c8y_prefix,
-        ))
+        )
+        .expect("Topic must have been valid as the external id is pre-validated");
+        Ok(topic)
+    }
+
+    pub fn parent_xid_if_not_main_device(
```

Comment:
I got it.

Option 2 sounds the correct choice (because moving away the c8y details under c8y_api).

However, it would be better to first stabilize this PR and add such improvements only in a second step.
For now, simply add this note as a doc comment.

Reply:
Resolved by 21c694f